docs/Architecture/SW-Upgrade.md

```diff
@@ -0,0 +1,115 @@
+# SW Upgrade
+
+Table of contents:
+
+- [Setup SW upgrade repository](#setup-sw-upgrade-repository)
+- [Apply SW upgrade](#apply-sw-upgrade)
+- [SW upgrade single ISO bundle structure](#sw-upgrade-single-iso-bundle-structure)
+- [Delete SW upgrade repository](#delete-sw-upgrade-repository)
+
+
+## Setup SW upgrade repository
+
+The common structure of the `set_swupgrade_repo` command:
+```bash
+provisioner set_swupgrade_repo </path/to/single/iso/bundle> --hash="<hash_type>:<hash> <filename>"  --hash-type="<hash_type>" \ 
+--username=<provisioner_user> --password=<provisioner_user_password>`
```

Comment:
@dmitrydb I think we don't need mentions of authentication options in the document. It should be a part of other docs, and here it mostly shifts a reader's focus.

Reply:
Done, I added the `References` section and added references across the doc. I also added a small section about a high-level logic overview of SW upgrade.